File: regexlib/2021-5-15/python_re2_test_file/regexlib_1114.py
# 1114
# ^p(ost)?[ |\.]*o(ffice)?[ |\.]*(box)?[ 0-9]*[^[a-z ]]*
# POLYNOMIAL
# nums:5
# POLYNOMIAL AttackString:"po"+" "*5000+"! _1_POA(i)"

import re2 as re
from time import perf_counter

regex = """^p(ost)?[ |\.]*o(ffice)?[ |\.]*(box)?[ 0-9]*[^[a-z ]]*"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "po" + " " * i * 10000 + "! _1_POA(i)"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *10000}: took {DURATION} seconds!")

File: regexlib/2021-5-15/javascript_test_file/regexlib_1114.js
/* 1114
 * ^p(ost)?[ |\.]*o(ffice)?[ |\.]*(box)?[ 0-9]*[^[a-z ]]*
 * POLYNOMIAL
 * nums:3
 * POLYNOMIAL AttackString:"po"+" "*5000+"! _1_POA(i)"
 */
var REGEX = /^p(ost)?[ |\.]*o(ffice)?[ |\.]*(box)?[ 0-9]*[^[a-z ]]*/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = 'po' + ' '.repeat(i*10000) + '! _1_POA(i)'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 10000 + ": " + time_cost+" ms")
}